### ISSUE TITLE
Feature Request: Port CRAN's fuzzyjoin Funtions to dplyr

### ISSUE THREAD

**Issue by jtelleriar** [tidyverse/dplyr#3293]
I would suggest porting some of CRAN's fuzzyjoin package functions to dplyr (With proper Contribution Acknowledgement):

https://cran.r-project.org/web/packages/fuzzyjoin/index.html

Of special interest is "fuzzy_join" function, which allows to do SQL queries such as these:
```
SELECT
    REPLACE(PATH, 'W:\\', '\\\\naswoav.usr.corp.gamesa.es\\WO_AV_DATA\\') AS PATH
  FROM
    dispoin.t_DISPOIN_DIR AS A
    INNER JOIN dispoin.t_scadas AS B
      ON
        A.PATH LIKE CONCAT('%', ScadaCode, '%')
        AND (A.PATH LIKE '%2016%' OR A.PATH LIKE '%2017%')
        -- También se puede usar  A.PATH REGEXP '2016|2017';
        AND A.PATH NOT LIKE '%Data_10M%'
        AND A.PATH NOT LIKE '%rejectAL%'
        AND A.PATH NOT LIKE '%REPROCESSED_FILES%'
        AND A.PATH NOT LIKE '%GA677_201308_AL.csv.s3542017.zip%';
```

**Comment by krlmlr** [comment]
Thanks. Are you suggesting to run the fuzzy join directly on the database?

**Comment by jtelleriar** [comment]
Either in database or in R itself.

However, when you query an SQL database I guess you could always do a cheat like this:
```
t_DISPOIN_DIR <- tbl(con, "t_DISPOIN_DIR")

t_DISPOIN_DIR %>%
sql(
"
SELECT
    REPLACE(PATH, 'W:\\', '\\\\naswoav.usr.corp.gamesa.es\\WO_AV_DATA\\') AS PATH
  FROM
    dispoin.t_DISPOIN_DIR AS A
    INNER JOIN dispoin.t_scadas AS B
      ON
        A.PATH LIKE CONCAT('%', ScadaCode, '%')
        AND (A.PATH LIKE '%2016%' OR A.PATH LIKE '%2017%')
        -- También se puede usar  A.PATH REGEXP '2016|2017';
        AND A.PATH NOT LIKE '%Data_10M%'
        AND A.PATH NOT LIKE '%rejectAL%'
        AND A.PATH NOT LIKE '%REPROCESSED_FILES%'
        AND A.PATH NOT LIKE '%GA677_201308_AL.csv.s3542017.zip%';
")
```

Although it's a bit like a cheat.
Juan

**Comment by krlmlr** [comment]
*fuzzyjoin* seems to be removed from CRAN, are you aware of a successor?

**Comment by jtelleriar** [comment]
At the company where I work we have GitHub Webpage Blocked, and I cannot check properly this issue.

The point is that if the package has been removed from CRAN, there is even more reason to fork *some* of its functions (as they are) and move them to mainstream dplyr.

I checked them out in the past and they are properly written.

The fuzzyjoin GitHub Repository is at the following link:

https://github.com/dgrtwo/fuzzyjoin

I'll check it out when I arrive at home, though these days I am little busy learning the H2O.ai framework, and doing a couple of Cheatsheets for using its machine learning functions.

Juan
